stdlib/public/TensorFlow/Tensor.swift

```diff
@@ -506,6 +507,7 @@ public extension Tensor {
   @inlinable
   var shape: TensorShape {
     @inline(__always)
+    @_semantics("autodiff.nonvarying")
```

Comment:
Why is this tag necessary? I'd expect `x.withoutDerivative().shape` to just work.

Reply:
Oh I see, maybe you wanted to make this a known 0-derivative function. In that case you can use `@noDerivative`. If that attribute doesn't work yet, we should make it work. Feel free to do that in a separate patch.

Reply:
It does just work. It requires staging with the layers API though. Either I need to go: add withoutDerivative(), update swift APIs, this PR. or this and then delete these later, or some way to commit it all atomically.